src/Nethermind/Nethermind.Init/Steps/EthereumStepsManager.cs

```diff
@@ -201,6 +177,75 @@ private void RunOneRoundOfInitialization(CancellationToken cancellationToken)
             return step;
         }
 
+        private Task RunPluginBeforeStep(IStep step)
+        {
+            Stopwatch stopwatch = Stopwatch.StartNew();
+            return Task.WhenAll(_api.Plugins.Select(p => p.BeforeStep(step.Name).ContinueWith(t =>
```

Comment:
I would execute steps in simple for with async/await. Plugins have order and they can relate one to another, lets not introduce parallelism here.